### PR TITLE
Milestone 3: Add Functionality to the checkbox and Clear all completed button

### PR DESCRIPTION
**Pull request by PrantoshB**
- Add functionality to the checkbox
- Clear all complete button filters the array and regenerates the list
- Use LocalStorage to store and fetch updated data
- Fix Linters Errors
- Follow GitHub Flow

### CODE REVIEW

File: src/modules/crud.js [hunk 1]
```diff
@@ -30,19 +31,60 @@ InputField.addEventListener("keypress", (e) => {
...
+  else {
+    specList.style.textDecoration = 'none'
+    specList.style.color = 'inherit'
+    let storedData = localStorage.getItem("To-Do");
+    ToDoData = JSON.parse(storedData);
+    ToDoData[index].completed = false;
+    localStorage.setItem("To-Do", JSON.stringify(ToDoData));
+    addToList();
+  }
+}
```
Comment (jerryowusu) [inline src/modules/crud.js:85]: I am really impressed with your project so, neat and well organized code plus everything is working fine :1st_place_medal: However, the project requires you to implement a function for the `clear all completed` button using the `array.filter` method. :wink:  Kindly update that to make your project complete and in line with the project requirements. :pray: 

Screenshot from 2022-12-09 07-05-58

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @PrantoshB  

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:


no-we-were-so-close-pauline-turner-brooks


## Highlights
Neat and well organized code :+1: 
Great UI design :+1: 
Everything is working fine :100: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/crud.js (hunk 1 above)
Comment (PrantoshB) [reply src/modules/crud.js:85]: @jerryowusu kindly look at line 141. The filter method is already there.

**Review by sinansevgi — COMMENTED** [review]
Hello @PrantoshB  👋,

_This is an additional review as a result of the second opinion request._

I can confirm that you already using the filter method to filter completed tasks 👍 Sorry for this inconvenience

As this was the only reason that your project was not approved **it will be approved** as soon as possible.
Please ask for another review in your Dashboard and the **next Code Reviewer** will mark it as **approve** right away :)

Keep rocking 🚀!

**Review by frank1738 — APPROVED** [review]
Hi @PrantoshB , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Github workflow
✅ No linter error
✅ README is okay
✅ Good PR Description 👍 
✅ App is working as expected 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**